packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -329,6 +334,31 @@ export class DataTable {
     this.columnOrdering(newColumns);
   }
 
+  /**
+   * selectAllRows method we can use to select/unselect rows in the table
+   * @param checked denotes if we want to check or uncheck the rows
+   */
+  @Method()
+  async selectAllRows(checked = true) {
+    if (this.isAllSelectable) {
+      const selectAll: HTMLFwCheckboxElement = this.el.shadowRoot.querySelector(
```

Comment:
instead of using manually querying the dom, can we have State props that can be used to rerender the templates? 

How will the parent app know what rows are selected?

Reply:
I agree. Making UI changes based on State/Prop changes is more appropriate. I will make necessary changes and update the PR. Will also return selected row IDs from this function for parent to know what are the selected rows.

Reply:
Updated as suggested.